src/main/java/seedu/address/model/task/DistinctTaskList.java

```diff
@@ -88,6 +89,19 @@ public void replaceTask(Task target, Task editedTask, boolean isSameTask) throws
         taskList.set(index, editedTask);
     }
 
+    /**
+     * Unlinks all tasks that are currently linked to {@code exam}.
+     * @param exam
+     */
+    public void unlinkTasksFromExam(Exam exam) {
+        taskList.forEach(task -> {
```

Comment:
Maybe you could add null check for exam here.

Reply:
I agree, will update accordingly.